bundles/org.openhab.binding.broadlink/src/main/java/org/openhab/binding/broadlink/handler/BroadlinkBaseThingHandler.java

```diff
@@ -0,0 +1,293 @@
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.openhab.core.util.HexUtils;
+import org.slf4j.Logger;
+
+/**
+ * Abstract superclass of all supported Broadlink devices.
+ *
+ * @author John Marshall/Cato Sognen - Initial contribution
+ */
+@NonNullByDefault
+public abstract class BroadlinkBaseThingHandler extends BaseThingHandler implements DeviceRediscoveryListener {
+
+    private static final String INITIAL_DEVICE_ID = "00000000";
+
+    @Nullable
+    private RetryableSocket socket;
+
+    @Nullable // Can be injected for test purposes
+    private NetworkTrafficObserver networkTrafficObserver;
+
+    private int count;
+
+    protected BroadlinkDeviceConfiguration thingConfig;
+    protected final ThingLogger thingLogger;
+    @Nullable
+    private ScheduledFuture<?> refreshHandle;
+    private boolean authenticated = false;
+    // These get handed to us by the device after successful authentication:
+    private byte[] deviceId;
+    private byte[] deviceKey;
+
+    public BroadlinkBaseThingHandler(Thing thing, Logger logger) {
+        super(thing);
+        this.thingLogger = new ThingLogger(thing, logger);
+        this.thingConfig = getConfigAs(BroadlinkDeviceConfiguration.class);
+        count = (new Random()).nextInt(65535);
+
+        thingLogger.logInfo(String.format("constructed: resetting deviceKey to '%s', length %d",
+                BroadlinkBindingConstants.BROADLINK_AUTH_KEY, BroadlinkBindingConstants.BROADLINK_AUTH_KEY.length()));
+        thingLogger.logInfo("(HINT: this should start '0976', end '8b02' and have length 32)");
+        this.deviceId = HexUtils.hexToBytes(INITIAL_DEVICE_ID);
+        this.deviceKey = HexUtils.hexToBytes(BroadlinkBindingConstants.BROADLINK_AUTH_KEY);
+
+        this.socket = new RetryableSocket(thingConfig, thingLogger);
+    }
+
+    // For test purposes
+    void setSocket(RetryableSocket socket) {
+        this.socket = socket;
+    }
+
+    void setNetworkTrafficObserver(NetworkTrafficObserver networkTrafficObserver) {
+        this.networkTrafficObserver = networkTrafficObserver;
+    }
+
+    private boolean hasAuthenticated() {
+        return authenticated;
+    }
+
+    public void initialize() {
+        thingLogger.logDebug("initializing polling");
+
+        updateItemStatus();
+
+        if (thingConfig.getPollingInterval() != 0) {
+            refreshHandle = scheduler.scheduleWithFixedDelay(new Runnable() {
+
+                public void run() {
+                    updateItemStatus();
+                }
+            }, 1L, thingConfig.getPollingInterval(), TimeUnit.SECONDS);
+        }
+    }
+
+    public void thingUpdated(Thing thing) {
+        thingLogger.logDebug("thingUpdated");
```

Comment:
Please read the section on logging here
https://www.openhab.org/docs/developer/guidelines.html#f-logging
If the thing is updated, the event.log already gets more info sent to it, this will increase the load on the system for no reason.